lib/firmata.js

```diff
@@ -209,20 +211,28 @@ SYSEX_RESPONSE[CAPABILITY_RESPONSE] = function(board) {
 
   // Only create pins if none have been previously created on the instance.
   if (!board.pins.length) {
+    var pin = {};
     for (var i = 2, n = 0; i < board.currentBuffer.length - 1; i++) {
       if (board.currentBuffer[i] === 127) {
-        board.pins.push({
-          supportedModes: supportedModes(capability),
-          mode: undefined,
-          value: 0,
-          report: 1,
-        });
+        pin.supportedModes = supportedModes(capability);
+        pin.mode = undefined;
+        pin.value = 0;
+        pin.report = 1;
+        board.pins.push(pin);
+        pin = {};
         capability = 0;
         n = 0;
         continue;
       }
       if (n === 0) {
         capability |= (1 << board.currentBuffer[i]);
```

Comment:
although it's probably not good practice to initialize a variable in a block where that variable is also used out side that block (even though it works with var)